src/invidious/channels/videos.cr

```diff
@@ -171,4 +91,100 @@ module Invidious::Channel::Tabs
 
     return items, next_continuation
   end
+
+  # -------------------
+  #  C-tokens
+  # -------------------
+
+  private def sort_options_videos_short(sort_by : String)
+    case sort_by
+    when "newest"  then return 4_i64
+    when "popular" then return 2_i64
+    when "oldest"  then return 5_i64
+    else                return 4_i64 # Fallback to "newest"
+    end
+  end
+
+  # Generate the initial "continuation token" to get the first page of the
+  # "videos" tab. The following page requires the ctoken provided in that
+  # first page, and so on.
+  private def make_initial_videos_ctoken(ucid : String, sort_by = "newest")
+    object = {
+      "15:embedded" => {
+        "2:string" => "\n$00000000-0000-0000-0000-000000000000",
```

Comment:
Weird, shouldn't this be an embedded with a 1:string in it?

Reply:
I have absolutely no idea why this one is different like that, but that's what the official client sends, sooooo....

Reply:
Actually this is interesting, they are the same!
the `\n` is basically a decimal `10`, which is 1010 binary. That is a field number 1, and a wire type 2 (length-delimited)
then the `$` is a decimal `36`, which is exactly the length of `00000000-0000-0000-0000-000000000000`.
So both objects end up being encoded into the same data.
I think having it like the other one (embedded object with a subfield number 1) would be better for clarity (and could also allow you to refactor / reuse code)

Reply:
Ohh, you're right! That's likely a weird bug of our parser `Protodec`.

> having it like the other one (embedded object with a subfield number 1) would be better for clarity (and could also allow you to refactor / reuse code)

Yeah, that's indeed much less cryptic that way! However, I'll keep the two functions separate to not need an extra parameter and a string interpolation.